core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java

```diff
@@ -94,27 +100,34 @@ public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
     public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
       refreshAdapter = false;
       if (item.getItemId() == R.id.menu_context_delete) {
-        fullHistory.removeAll(deleteList);
-        for (HistoryListItem history : deleteList) {
-          int position = historyList.indexOf(history);
+        alertDialogShower.show(KiwixDialog.DeleteHistoryDialog.INSTANCE,
+          (Function0<Unit>) () -> {
+            fullHistory.removeAll(deleteList);
+            for (HistoryListItem history : deleteList) {
+              int position = historyList.indexOf(history);
               /*
               Delete the current category header if there are no items after the current one or
               if the item being removed is between two category headers.
                */
-          if (position - 1 >= 0 && historyList.get(position - 1) == null &&
-            (position + 1 >= historyList.size() ||
-              (position + 1 < historyList.size() && historyList.get(position + 1) == null))) {
-            historyList.remove(position - 1);
-            historyAdapter.notifyItemRemoved(position - 1);
-          }
-          position = historyList.indexOf(history);
-          historyList.remove(history);
-          historyAdapter.notifyItemRemoved(position);
-          historyAdapter.notifyItemRangeChanged(position, historyAdapter.getItemCount());
-        }
-        presenter.deleteHistory(new ArrayList<>(deleteList));
-        mode.finish();
+              if (position - 1 >= 0 && historyList.get(position - 1) == null &&
+                (position + 1 >= historyList.size() ||
+                  (position + 1 < historyList.size() && historyList.get(position + 1) == null))) {
+                historyList.remove(position - 1);
+                historyAdapter.notifyItemRemoved(position - 1);
+              }
+              position = historyList.indexOf(history);
+              historyList.remove(history);
+              historyAdapter.notifyItemRemoved(position);
+              historyAdapter.notifyItemRangeChanged(position, historyAdapter.getItemCount());
+            }
+            presenter.deleteHistory(new ArrayList<>(deleteList));
+            mode.finish();
+            return Unit.INSTANCE;
+          });
+
         return true;
+
```

Comment:
this empty space probably violates our codestyle, press `ctrl+alt+l` in this file, I generally advocate for developers to bind a macro that saves, formats & optimises, I use `ctrl+s` for example